Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/TextBoxLikePropertyCtrl.hxx

```diff
@@ -0,0 +1,148 @@
+              Derived* GetDerived()       { return static_cast<      Derived*>(this); }
+
+        struct DeferredSetValue
+        {
+            T prevValue;
+            T value;
+        };
+
+        //! If a value is editing, but not by the user, and the user is currently editing the value,
+        //! avoid overwriting their work, until they finish editing
+        AZStd::optional<DeferredSetValue> m_deferredExternalValue;
+    };
+
+    template<class Derived, class T, class Traits>
+    void DeferredTextboxLikeEdit<Derived, T, Traits>::setValueFromSystem(const T& newValue)
+    {
+        if (shouldReallySetValue(Traits::get(GetDerived()), newValue, Traits::isUserEditing(GetDerived())))
+        {
+            Traits::set(GetDerived(), newValue);
+        }
+    }
+
+    template<class Derived, class T, class Traits>
+    template<class... Args>
+    void DeferredTextboxLikeEdit<Derived, T, Traits>::onTextBoxLikeEditingFinished(Args&&... args)
+    {
+        if (auto newValue = getDeferredValuedToSet())
+        {
+            Traits::set(GetDerived(), *newValue);
+        };
+
+        Traits::emitEditingFinished(GetDerived(), AZStd::forward<Args>(args)...);
+    }
+
+    template<class Derived, class T, class Traits>
+    bool DeferredTextboxLikeEdit<Derived, T, Traits>::shouldReallySetValue(const T& prevValue, const T& newValue, bool isUserEditing)
+    {
+        if constexpr (AZStd::is_floating_point_v<T>)
+        {
+            // Nothing to do if the value is not actually changed
+            if (AZ::IsCloseMag(prevValue, newValue, AZStd::numeric_limits<T>::epsilon()))
+            {
+                return false;
+            }
+        }
+
+        // If the spin box currently has focus, the user is editing it, so we should not
+        // change the value from non-user input while they're in the middle of editing
+        if (isUserEditing)
+        {
+            auto& deferredValue = m_deferredExternalValue.emplace();
+            deferredValue.value = newValue;
+            deferredValue.prevValue = prevValue;
+            return false;
+        }
+
+        return true;
+    }
+
+    template<class Derived, class T, class Traits>
+    AZStd::optional<T> DeferredTextboxLikeEdit<Derived, T, Traits>::getDeferredValuedToSet(const T& value, [[maybe_unused]] bool isUserEditing)
+    {
+        if (m_deferredExternalValue)
+        {
+            DeferredSetValue deferredValue = *m_deferredExternalValue;
+            m_deferredExternalValue.reset();
+
+            if (value == deferredValue.prevValue)
+            {
+                return deferredValue.value;
+            }
+        }
+
+        return {};
+    }
+
+    template<class Derived, class T, class Traits>
+    AZStd::optional<T> DeferredTextboxLikeEdit<Derived, T, Traits>::getDeferredValuedToSet()
+    {
+        return getDeferredValuedToSet(Traits::get(GetDerived()), Traits::isUserEditing(GetDerived()));
+    }
+
+
+AZ_PUSH_DISABLE_DLL_EXPORT_MEMBER_WARNING
+
+}
```

Comment:
nit: closing namespace

Reply:
(clang-format will do this for you 🙂 You can invoked from VS - ping me if you need more info)